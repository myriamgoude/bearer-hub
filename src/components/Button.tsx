import * as React from 'react'
import styled from '@emotion/styled'
import Link from './Link'

import { colors } from '../styles/variables'

const StyledButton = styled.button<{ primary: boolean }>`
  background-color: ${props => (props.primary ? colors.accent : colors.dark)};
  color: ${props => (props.primary ? colors.dark : colors.white)};
  padding: 0.5rem 1rem;
  margin-right: 1rem;
  border-radius: 4px;
`

interface IButtonProps {
  link: string
  text: string
  primary?: boolean
  className?: string
}

const Button = (props: IButtonProps) => (
  <Link to={props.link}>
    <StyledButton primary={props.primary || false}>{props.text}</StyledButton>
  </Link>
)

export default Button
