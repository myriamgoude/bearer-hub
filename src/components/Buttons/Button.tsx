import * as React from 'react'
import styled from '@emotion/styled'

import Link from '../Link/Link'

import styles from './Button.style'

interface IButtonProps {
  link: string
  text: string
  type?: string
  small?: boolean
  medium?: boolean
  primary?: boolean
  secondary?: boolean
  save?: boolean
  className?: string
  onClick?: any
  trackLink?: boolean
  trackingAction?: string
  trackingOptions?: any
}

const Button = (props: IButtonProps) => {
  const RegularButton = styled.button`
    ${styles.buttonStyle}

    ${props.primary && styles.primaryButton}
    ${props.secondary ? styles.secondaryButton : styles.primaryButton}
    ${props.save && styles.saveButton}
    ${
      props.small
        ? styles.smallButtonDimensions
        : props.medium
        ? styles.mediumButtonDimensions
        : styles.regularButtonDimensions
    }
  `
  const ButtonLink = styled(Link)`
    ${styles.linkStyle}
  `
  return (
    <ButtonLink
      to={props.link}
      onClick={props.onClick}
      trackLink={props.trackLink}
      trackingAction={props.trackingAction}
      trackingOptions={props.trackingOptions}
    >
      <RegularButton>{props.text}</RegularButton>
    </ButtonLink>
  )
}

export default Button
