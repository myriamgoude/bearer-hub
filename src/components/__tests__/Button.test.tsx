import * as React from 'react'
import renderer from 'react-test-renderer'
import Button from '../Buttons/Button'

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Button link="/foo" text="hello" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders primary colours', () => {
    const tree = renderer.create(<Button primary link="/foo" text="hello" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
