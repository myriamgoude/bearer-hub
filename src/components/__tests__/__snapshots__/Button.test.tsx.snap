// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.emotion-1 + a {
  margin-left: 0.5rem;
}

.emotion-0 {
  border-radius: 4px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  cursor: pointer;
  font-weight: 600;
  background-color: #FFC400;
  color: #030D36;
  height: 48px;
  line-height: 48px;
  padding: 0 1.5rem;
}

.emotion-0:hover {
  background-color: #E9B300;
}

<a
  className="emotion-1"
  href="/foo"
>
  <button
    className="emotion-0"
  >
    hello
  </button>
</a>
`;

exports[`Header renders primary colours 1`] = `
.emotion-1 + a {
  margin-left: 0.5rem;
}

.emotion-0 {
  border-radius: 4px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  cursor: pointer;
  font-weight: 600;
  background-color: #FFC400;
  color: #030D36;
  background-color: #FFC400;
  color: #030D36;
  height: 48px;
  line-height: 48px;
  padding: 0 1.5rem;
}

.emotion-0:hover {
  background-color: #E9B300;
}

.emotion-0:hover {
  background-color: #E9B300;
}

<a
  className="emotion-1"
  href="/foo"
>
  <button
    className="emotion-0"
  >
    hello
  </button>
</a>
`;
